x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/CertificateTool.java

```diff
@@ -818,17 +831,26 @@ void generateAndWriteSignedCertificates(Path output, boolean writeZipFile, Optio
                 CertificateInformation certificateInformation = certs.iterator().next();
                 CertificateAndKey pair = generateCertificateAndKey(certificateInformation, caInfo, keySize, days);
                 fullyWriteFile(output, stream -> writePkcs12(output.getFileName().toString(), stream,
-                    certificateInformation.name.originalName, pair, caInfo.certAndKey.cert, outputPassword, terminal));
+                    certificateInformation.name.originalName, pair,
+                    caInfo == null ? null : caInfo.certAndKey.cert, outputPassword, terminal));
             }
         }
 
         private CertificateAndKey generateCertificateAndKey(CertificateInformation certificateInformation, CAInfo caInfo,
                                                             int keySize, int days) throws Exception {
             KeyPair keyPair = CertGenUtils.generateKeyPair(keySize);
-            Certificate certificate = CertGenUtils.generateSignedCertificate(certificateInformation.name.x500Principal,
-                getSubjectAlternativeNamesValue(certificateInformation.ipAddresses, certificateInformation.dnsNames,
-                    certificateInformation.commonNames),
-                keyPair, caInfo.certAndKey.cert, caInfo.certAndKey.key, days);
+            Certificate certificate;
+            if (caInfo != null) {
+                certificate = CertGenUtils.generateSignedCertificate(certificateInformation.name.x500Principal,
+                    getSubjectAlternativeNamesValue(certificateInformation.ipAddresses, certificateInformation.dnsNames,
+                        certificateInformation.commonNames),
+                    keyPair, caInfo.certAndKey.cert, caInfo.certAndKey.key, days);
+            } else {
+                certificate = CertGenUtils.generateSignedCertificate(certificateInformation.name.x500Principal,
+                    getSubjectAlternativeNamesValue(certificateInformation.ipAddresses, certificateInformation.dnsNames,
+                        certificateInformation.commonNames),
+                    keyPair, null, null, true, days, null);
```

Comment:
```suggestion
                    keyPair, null, null, false, days, null);
```
The `true` means that this is a CA, but then `--self-signed` is just doing the same thing as `certutil ca`, but it should be generating a non ca, self-signed cert,